test/ngRoute/routeSpec.js

```diff
@@ -901,223 +901,632 @@ describe('$route', function() {
+          $rootScope.$digest();
+          expect(controller).toHaveBeenCalled();
+          expect(templateFn).toHaveBeenCalled();
+          expect($location.path()).toEqual('/redirect/to/undefined');
+          dealoc(element);
+        });
       });
     });
 
+    describe('via `resolveRedirectTo`', function() {
+      var $compile;
+      var $location;
+      var $rootScope;
+      var $route;
 
-    it('should properly interpolate optional and eager route vars ' +
-       'when redirecting from path with trailing slash', function() {
-      module(function($routeProvider) {
-        $routeProvider.when('/foo/:id?/:subid?', {templateUrl: 'foo.html'});
-        $routeProvider.when('/bar/:id*/:subid', {templateUrl: 'bar.html'});
+      beforeEach(module(function() {
+        return function(_$compile_, _$location_, _$rootScope_, _$route_) {
+          $compile = _$compile_;
+          $location = _$location_;
+          $rootScope = _$rootScope_;
+          $route = _$route_;
+        };
+      }));
+
+
+      it('should be ignored if `redirectTo` is also present', function() {
+        var newUrl;
+        var getNewUrl = function() { return newUrl; };
+
+        var resolveRedirectToSpy = jasmine.createSpy('resolveRedirectTo').and.returnValue('/bar');
+        var redirectToSpy = jasmine.createSpy('redirectTo').and.callFake(getNewUrl);
+        var templateSpy = jasmine.createSpy('template').and.returnValue('Foo');
+
+        module(function($routeProvider) {
+          $routeProvider.
+            when('/foo', {
+              resolveRedirectTo: resolveRedirectToSpy,
+              redirectTo: redirectToSpy,
+              template: templateSpy
+            }).
+            when('/bar', {template: 'Bar'}).
+            when('/baz', {template: 'Baz'});
+        });
+
+        inject(function() {
+          newUrl = '/baz';
+          $location.path('/foo');
+          $rootScope.$digest();
+
+          expect($location.path()).toBe('/baz');
+          expect($route.current.template).toBe('Baz');
+          expect(resolveRedirectToSpy).not.toHaveBeenCalled();
+          expect(redirectToSpy).toHaveBeenCalled();
+          expect(templateSpy).not.toHaveBeenCalled();
+
+          redirectToSpy.calls.reset();
+
+          newUrl = undefined;
+          $location.path('/foo');
+          $rootScope.$digest();
+
+          expect($location.path()).toBe('/foo');
+          expect($route.current.template).toBe(templateSpy);
+          expect(resolveRedirectToSpy).not.toHaveBeenCalled();
+          expect(redirectToSpy).toHaveBeenCalled();
+          expect(templateSpy).toHaveBeenCalled();
+        });
       });
 
-      inject(function($location, $rootScope, $route) {
-        $location.path('/foo/id1/subid2/');
-        $rootScope.$digest();
 
-        expect($location.path()).toEqual('/foo/id1/subid2');
-        expect($route.current.templateUrl).toEqual('foo.html');
+      it('should redirect to the returned url', function() {
```

Comment:
Isn't this really just a special case of the `should support returning a promise` test further down?
Perhaps it would be clearer to keep them together?